solutions/devsprint-matheus-reis-7/FinanceApp/Modules/ActivityDetails/ActivityDetailsViewController.swift

```diff
@@ -6,19 +6,77 @@
 //
 
 import UIKit
+import RxCocoa
+import RxSwift
 
-class ActivityDetailsViewController: UIViewController {
+final class ActivityDetailsViewController: UIViewController {
 
-    lazy var activityDetailsView: ActivityDetailsView = {
+    // MARK: - UIView properties
+    private let activityDetailsView: ActivityDetailsViewProtocol
 
-        let activityDetailsView = ActivityDetailsView()
-        activityDetailsView.delegate = self
-        return activityDetailsView
-    }()
+    // MARK: - DataSource / UseCase dependencies
+    private let service: FinanceService
+
+    // MARK: - Reactive Properties
+    private let disposeBag: DisposeBag
+    private let activityDetailsObservable = BehaviorRelay<ActivityDetails>(
+        value: .init(name: "",
+                     price: 0,
+                     category: "",
+                     time: "")
+    )
+
+    // MARK: - Initializers
+    // TODO: Abstract FinanceService as ServiceProtocol
+    init(activityDetailsView: ActivityDetailsViewProtocol = ActivityDetailsView(),
+         service: FinanceService = FinanceService(),
+         disposeBag: DisposeBag = DisposeBag()) {
+        self.activityDetailsView = activityDetailsView
+        self.service = service
+        self.disposeBag = disposeBag
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) { nil }
 
+    // MARK: - UIViewController lifecycle methods
     override func loadView() {
         self.view = activityDetailsView
     }
+
+    override func viewDidLoad() {
+        activityDetailsView.delegate = self
+        fetchDataView()
+        bindObservables()
+    }
+}
+
+// MARK: -
+private extension ActivityDetailsViewController {
+
+    private func fetchDataView() {
+        service.fetchActivityDetails().subscribe(onSuccess: { [weak self ] activityDetails in
+            guard let self = self else { return }
+            self.activityDetailsObservable.accept(activityDetails)
+        }, onFailure: { failure in
+            // TODO: What should be the approach on errors?
+            print("failure:", failure.localizedDescription)
+        }).disposed(by: disposeBag)
+    }
+
+}
+
+// MARK: - Observables binding
+private extension ActivityDetailsViewController {
+
+    private func bindObservables() {
+        activityDetailsObservable
+            .asDriver()
+            .drive {
+                self.activityDetailsView.show(viewModel: $0)
```

Comment:
Nice usage of a `Driver`, keep that in mind: when you work with reactive approach it's always better to 'drive', i.e redirect the values you had from a stream to a subject 🥇